native/cocos/platform/android/AndroidPlatform.cpp

```diff
@@ -25,79 +25,460 @@
 
 #include <thread>
 
+#include "base/memory/Memory.h"
 #include "platform/android/AndroidPlatform.h"
+#include "platform/BasePlatform.h"
 #include "platform/java/jni/glue/JniNativeGlue.h"
 #include "platform/java/modules/Accelerometer.h"
 #include "platform/java/modules/Battery.h"
 #include "platform/java/modules/Network.h"
 #include "platform/java/modules/SystemWindow.h"
 #include "platform/java/modules/Vibrator.h"
+#include "cocos/bindings/event/CustomEventTypes.h"
 #include "modules/Screen.h"
 #include "modules/System.h"
+#include "java/jni/log.h"
+#include "java/jni/JniHelper.h"
 
-namespace cc {
-AndroidPlatform::AndroidPlatform() {
-    _jniNativeGlue = JNI_NATIVE_GLUE();
-}
+#include "game-activity/GameActivity.cpp"
+#include "game-text-input/GameTextInput.cpp"
+#include "bindings/event/EventDispatcher.h"
 
-int AndroidPlatform::init() {
-    registerInterface(std::make_shared<Accelerometer>());
-    registerInterface(std::make_shared<Battery>());
-    registerInterface(std::make_shared<Network>());
-    registerInterface(std::make_shared<Screen>());
-    registerInterface(std::make_shared<System>());
-    registerInterface(std::make_shared<SystemWindow>());
-    registerInterface(std::make_shared<Vibrator>());
-    return 0;
+extern "C" {
+#include "game-activity/native_app_glue/android_native_app_glue.c"
 }
 
-int AndroidPlatform::getSdkVersion() const {
-    return _jniNativeGlue->getSdkVersion();
-}
+#include "paddleboat/paddleboat.h"
 
-int32_t AndroidPlatform::run(int argc, const char **argv) {
-    std::thread mainLogicThread([this, argc, argv]() {
-        waitWindowInitialized();
-        UniversalPlatform::run(argc, argv);
-        onDestory();
-    });
-    mainLogicThread.detach();
-    _jniNativeGlue->waitRunning();
-    return 0;
-}
 
-void AndroidPlatform::waitWindowInitialized() {
-    _jniNativeGlue->setRunning(true);
-    while (_jniNativeGlue->isRunning()) {
-        pollEvent();
-        NativeWindowType *wndHandle = _jniNativeGlue->getWindowHandler();
-        if (wndHandle != nullptr) {
-            break;
+#define ABORT_GAME { ALOGE("*** GAME ABORTING."); *((volatile char*)0) = 'a'; }
+
+#define MY_ASSERT(cond) { if (!(cond)) { ALOGE("ASSERTION FAILED: %s", #cond); \
+   ABORT_GAME; } }
```

Comment:
Why not use CC_ASSERT?

Reply:
Also in release mode?

Reply:
Got it. It is better not to use ASSERT, as it is confused with `assert`. And it is better not use `MY`, use a more meaningful name please.

Reply:
How about LOG_AND_ABORT?

Reply:
Or ABORT_IF.